apps/next/components/SignUpButton.tsx

```diff
@@ -0,0 +1,26 @@
+"use client";
+import { Skeleton } from "@/components/ui/skeleton";
+import { useUser, useClerk } from "@clerk/nextjs";
+import { useRouter } from "next/navigation";
+import Link from "next/link";
+import { Button } from "@/components/ui/button";
+
+export const SignUpButton = () => {
+  const { isLoaded, user } = useUser();
+
+  return (
+    <>
+      {!isLoaded ? (
```

Comment:
I understand why you used negated conditions here, but I suggest you separate these ternary conditions as it's not very readable.
```
if (!isLoaded) {
  // return skeleton component
}

if (!user) {
  // return signup button
}

return link to dashboard
```

Reply:
👍